### ISSUE TITLE
unmet peer vite@"^2.2.3 || ^3.0.0": found 4.0.0

### ISSUE THREAD

**Issue by n8sabes** [vanilla-extract-css/vanilla-extract#945]
### Describe the bug

 WARN  Issues with peer dependencies found
.
├─┬ @vanilla-extract/vite-plugin 3.7.0
│ └── ✕ unmet peer vite@"^2.2.3 || ^3.0.0": found 4.0.0

### Reproduction

https://github.com/vanilla-extract-css/vanilla-extract/blob/9312b66e5bd67942b7929a6b93e0ad2181b2e0ba/packages/vite-plugin/package.json#L27

### System Info

```shell
System:
    OS: macOS 13.0.1
    CPU: (8) arm64 Apple M1
    Memory: 73.55 MB / 16.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 19.0.1 - ~/.nvm/versions/node/v19.0.1/bin/node
    npm: 8.19.2 - ~/.nvm/versions/node/v19.0.1/bin/npm
  Browsers:
    Chrome: 108.0.5359.98
    Firefox Developer Edition: 108.0
    Safari: 16.1
```

### Used Package Manager

pnpm

### Validations

- [X] Check that there isn't [already an issue](https://github.com/vanilla-extract-css/vanilla-extract/issues) that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by wmaurer** [comment]
I have the same problem using `npm@8.19.2`, which will error when installing `@vanilla-extract/vite-plugin` in a new project where `vite@4.0.3` is being used (in this case using Nx and `@nrwl/vite@15.4.0`).

Once can override the unmet dependency using:

    npm install --save-dev @vanilla-extract/vite-plugin --legacy-peer-deps

From then on, all seems to work though, so my very simple test seems to show that this plugin is compatible with vite v4.

**Comment by maxlapides** [comment]
Here's another temporary solution: you can add this to your `package.json`:

```json
"overrides": {
  "@vanilla-extract/vite-plugin": {
    "vite": "^4"
  }
}
```

**Comment by yuckiymouse** [comment]
> Here's another temporary solution: you can add this to your `package.json`:
> 
> ```json
> "overrides": {
>   "@vanilla-extract/vite-plugin": {
>     "vite": "^4"
>   }
> }
> ```

doesn't work, sadly.

**Comment by yuckiymouse** [comment]
> I have the same problem using `npm@8.19.2`, which will error when installing `@vanilla-extract/vite-plugin` in a new project where `vite@4.0.3` is being used (in this case using Nx and `@nrwl/vite@15.4.0`).
> 
> One can override the unmet dependency using:
> 
> ```
> npm install --save-dev @vanilla-extract/vite-plugin --legacy-peer-deps
> ```
> 
> From then on, all seems to work though, so my very simple test seems to show that this plugin is compatible with vite v4.

this works for me!

**Comment by askoufis** [comment]
Fixed in #949
